libraries/EEPROM/src/EEPROM.h

```diff
@@ -174,7 +179,14 @@ struct EEPROMClass
     return 0x00;
   }
   EEPtr end() { return length(); } //Standards requires this to be the item after the last valid entry. The returned pointer is invalid.
-  uint16_t length() { return AP3_FLASH_EEPROM_SIZE; }
+  uint16_t length()
+  {
+    return allowedSize;
+  }
+  void setLength(uint16_t length)
+  {
+    allowedSize = length;
```

Comment:
Shouldn't this be something along the lines of:
`allowedSize = (length <= AP3_EEPROM_MAX_LENGTH) ? length : AP3_EEPROM_MAX_LENGTH;`

Otherwise I am guessing an over-range input will cause undefined behavior.